Framework/Core/include/Framework/BinningPolicy.h

```diff
@@ -0,0 +1,331 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#ifndef FRAMEWORK_BINNINGPOLICY_H
+#define FRAMEWORK_BINNINGPOLICY_H
+
+#include "Framework/HistogramSpec.h" // only for VARIABLE_WIDTH
+#include "Framework/ASoAHelpers.h"
+#include "Framework/Pack.h"
+#include "Framework/ArrowTypes.h"
+#include <optional>
+
+namespace o2::framework
+{
+
+namespace binning_helpers
+{
+template <typename... Cs>
+std::array<arrow::ChunkedArray*, sizeof...(Cs)> getArrowColumns(arrow::Table* table, pack<Cs...>)
+{
+  static_assert(std::conjunction_v<typename Cs::persistent...>, "BinningPolicy: only persistent columns accepted (not dynamic and not index ones");
```

Comment:
An important point: one cannot do event mixing / block combinations based on values in dynamic columns. Simply because they do not exist as real columns.